crates/task-impls/src/quorum_vote.rs

```diff
@@ -109,6 +111,16 @@ impl<TYPES: NodeType, S: Storage<TYPES> + 'static> HandleDepOutput
                         payload_commitment = Some(vid_payload_commitment);
                     }
                 }
+                HotShotEvent::VoteNow(_, vote_dependency_data) => {
+                    payload_commitment = Some(
+                        vote_dependency_data
+                            .quorum_proposal
+                            .block_header
+                            .payload_commitment(),
+                    );
+                    leaf = Some(vote_dependency_data.leaf.clone());
```

Comment:
@shenkeyao looks like this code gets the payload_commitment from one of 2 possible sources, not sure if this will change in your upcoming work, but I can shrink the payload size if we only care about one possible option.

Reply:
I don't think we need to set `payload_commitment` if the event is `VoteNow`. We want to check if the commitments from the other two events match, but if we get `VoteNow` we know we can vote immediately.

Reply:
I sent an update to remove the `payload_commitment` setting from here.